project/CelebornBuild.scala

```diff
@@ -913,6 +913,31 @@ object MRClientProjects {
       )
   }
 
+  def mrIt: Project = {
+    val hadoopVersion = "3.3.6"
```

Comment:
3.2.4?

Reply:
Here are some dependency conflicts.

1. celeborn relies on guava 14, and hadoop-common for 3.2 relies on guava 27, so whichever version of guava is used here will result in `NoSuchMethodError`.

<details>
  <summary>NoSuchMethodError</summary>

2023-11-10T11:28:08,905 ERROR [main] org.apache.celeborn.mapreduce.v2.app.MRAppMasterWithCeleborn: Error starting MRAppMaster
java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkArgument(ZLjava/lang/String;Ljava/lang/Object;)V
	at org.apache.hadoop.conf.Configuration.set(Configuration.java:1357) ~[hadoop-common-3.2.4.jar:?]
	at org.apache.hadoop.conf.Configuration.set(Configuration.java:1338) ~[hadoop-common-3.2.4.jar:?]
	at org.apache.celeborn.mapreduce.v2.app.MRAppMasterWithCeleborn.<init>(MRAppMasterWithCeleborn.java:73) ~[classes/:?]
	at org.apache.celeborn.mapreduce.v2.app.MRAppMasterWithCeleborn.main(MRAppMasterWithCeleborn.java:149) ~[classes/:?]
</details>

But 3.3 removes some guava dependencies and replaces them with a guava shade.

[[HADOOP-17098](https://issues.apache.org/jira/browse/HADOOP-17098)] Reduce Guava dependency in Hadoop source code

[[HADOOP-17288](https://issues.apache.org/jira/browse/HADOOP-17288)] Use shaded guava from thirdparty

2. Since MiniYARNCluster is started with YARN with some shaded packages, but MR depends on `hadoop-mapreduce-client-app` which uses hadoop jar without shaded, so here we have to put MR dependent jar at the top of MR task classpath using `mapreduce.job.user. classpath.first=true` and `job.addFileToClassPath` to put the MR dependent jar at the top of the MR task classpath.

<details>
  <summary>NoSuchMethodError</summary>
[info]   Cause: java.lang.NoSuchMethodError: org.apache.hadoop.yarn.webapp.WebApps$Builder.start(Lorg/apache/hadoop/yarn/webapp/WebApp;Lorg/apache/hadoop/shaded/org/eclipse/jetty/webapp/WebAppContext;)Lorg/apache/hadoop/yarn/webapp/WebApp;
[info]   at org.apache.hadoop.yarn.server.resourcemanager.ResourceManager.startWepApp(ResourceManager.java:1389)
[info]   at org.apache.hadoop.yarn.server.resourcemanager.ResourceManager.serviceStart(ResourceManager.java:1498)
[info]   at org.apache.hadoop.service.AbstractService.start(AbstractService.java:194)
[info]   at org.apache.hadoop.yarn.server.MiniYARNCluster.startResourceManager(MiniYARNCluster.java:376)
[info]   at org.apache.hadoop.yarn.server.MiniYARNCluster.access$300(MiniYARNCluster.java:129)
[info]   at org.apache.hadoop.yarn.server.MiniYARNCluster$ResourceManagerWrapper.serviceStart(MiniYARNCluster.java:500)
</details>

<details>
  <summary>NoSuchMethodError</summary>
[info]   Cause: java.lang.NoSuchMethodError: org.apache.hadoop.security.authentication.server.AuthenticationFilter.constructSecretProvider(Ljavax/servlet/ServletContext;Ljava/util/Properties;Z)Lorg/apache/hadoop/security/authentication/util/SignerSecretProvider;
[info]   at org.apache.hadoop.http.HttpServer2.constructSecretProvider(HttpServer2.java:799)
[info]   at org.apache.hadoop.http.HttpServer2.<init>(HttpServer2.java:685)
[info]   at org.apache.hadoop.http.HttpServer2.<init>(HttpServer2.java:129)
[info]   at org.apache.hadoop.http.HttpServer2$Builder.build(HttpServer2.java:478)
[info]   at org.apache.hadoop.yarn.webapp.WebApps$Builder.build(WebApps.java:372)
[info]   at org.apache.hadoop.yarn.webapp.WebApps$Builder.start(WebApps.java:465)
[info]   at org.apache.hadoop.yarn.server.resourcemanager.ResourceManager.startWepApp(ResourceManager.java:1389)
[info]   at org.apache.hadoop.yarn.server.resourcemanager.ResourceManager.serviceStart(ResourceManager.java:1498)
</details>

Reply:
Maybe we can upgrade Celeborn's guava to 27 to resolve this issue. HDFS 3.2.4 client can access HDFS 2.x clutster while HDFS 3.3.6 can not access HDFS 2.x cluster.

Reply:
Spark itself is compatible with Guava 27+, but Hive 2.3.9 does not. I suppose we don't need to enable spark-hive module to do test, thus upgrading Guava 27 and shading it in the client module makes sense to me.

Reply:
https://github.com/apache/incubator-celeborn/pull/2090

Reply:
After https://github.com/apache/incubator-celeborn/pull/2090, Celeborn no longer relies on a specific version of guava, and is compatible with guava 14/27/32.

what about bumping guava to the latest version `32.1.3-jre`, but setting guava to version `27.0-jre` when running MapReduce integration tests?

Reply:
#2090 is merged. How about now? Does this patch needs some modification?